### ISSUE TITLE
Values.yaml ignored when patching since bump to charity 0.3.10

### ISSUE THREAD

**Issue by msutter** [roboll/helmfile#1365]
Hi,

Related to #1360, I've just tested the pathing on the prometheus-operator chart.
So, now the manifests are rendered without an error :-) , but the resulting manifests seems to ignore the values.yaml file defined in the helmfile.yaml definition and only use the default values.yaml def from the original chart, which is ..... not usable.

This only happens when defining a patch in the helmfile.yaml definitions, so I think it is related to the bump of chartify to 0.3.10.
btw, thank you for the fast reaction and release you had on #1360 👍

**Comment by msutter** [comment]
@mumoshu I think this one is important. If I'm right, this could brake a lot of things, just by updating helmfile to the last release (0.122.1). Every release with a patch, would just ignore the custom values and use the defaults.

**Comment by mumoshu** [comment]
@msutter Thanks for reporting! I have not modified those parts of the chartify in the recent bump. Very odd..

**Comment by msutter** [comment]
@mumoshu mmm, maybe the issue is on my chair :-)
I'll try again on a fresh linux machine, as I did my testing on my machine (osx), and maybe some caching or other stuff is happening.
I'll report here the results when it's done

**Comment by msutter** [comment]
@mumoshu The behaviour is the same on linux :-(

I've noted that not all values are ignored but only the enabling/disabling of dependencies.

Here the related output of a release diff of the prometheus operator with patching:

```
....
I0720 09:30:48.353283    1752 chartify.go:255] using requirements.yaml:
 dependencies:
   - name: kube-state-metrics
     version: 2.4.*
     repository: https://kubernetes-charts.storage.googleapis.com/
     condition: kubeStateMetrics.enabled
   - name: prometheus-node-exporter
     version: 1.7.*
     repository: https://kubernetes-charts.storage.googleapis.com/
     condition: nodeExporter.enabled
   - name: grafana
     version: 4.0.*
     repository: https://kubernetes-charts.storage.googleapis.com/
     condition: grafana.enabled
...
```

In the values.yaml file I have disabled grafana:

```
...
grafana:
  enabled: false
...
```

But the conditions the requirements.yaml defined in my values.yaml definition seems to be ignored. All grafana related manifests are part of the final rendering.

Hope this helps

**Comment by mumoshu** [comment]
@msutter Thanks! That helped a lot. And I've confirmed that it works for me after #1362 and Helmfile v0.122.2.

Would you mind confirming that it works for you too, and there's no other issues on your env?

**Comment by msutter** [comment]
@mumoshu hey, thanks.

I still have some strange changes in the Prometheus-Operator chart related to additional PrometheusRules, but not sure what's happening there.

For my point of view, the patching is working well now.

I discovered an other issue with patching and 'helmfile list', I'll open a new issue for it
